oak_remote_attestation_verification/src/verify.rs

```diff
@@ -0,0 +1,81 @@
+//
+// Copyright 2023 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! Provides top-level verification based on evidence, endorsments and reference values.
+
+use crate::proto::oak::attestation::v1::{
+    endorsements, reference_values, CbEndorsements, CbReferenceValues, Endorsements, Evidence,
+    OakContainersEndorsements, OakContainersReferenceValues, OakRestrictedKernelEndorsements,
+    OakRestrictedKernelReferenceValues, ReferenceValues,
+};
+
+/// Verifies entire setup by forwarding to individual setup types.
+pub fn verify(
+    evidence: &Evidence,
+    endorsements: &Endorsements,
+    reference_values: &ReferenceValues,
+) -> anyhow::Result<()> {
```

Comment:
I think we need to create an `AttestationResults` proto for future-proofing (which for now will just cointain a `bool` value). But could be a separate PR

Reply:
Yes, if the result must be sent over the network. But we don't know yet. I think we should procrastinate this until the need comes up.

It would just be the return value of the top-level verify call. Internally, anyhow::Result is very handy.

Reply:
It's not just for over-the-network, but will also probably be useful / necessary for FFI.

Reply:
Done.